### DISCUSSION TITLE
Categories in search query should be case-insensitive

### DISCUSSION THREAD

**Discussion by laymonage** [community/community#4113] [Discussions] [answered]
When creating discussion categories, you cannot have more than one category with the same name (**case-insensitive**).

However, when searching discussions with categories in the query, the `category:` parameter seems to be case-sensitive. This shouldn't be the case, as it's impossible to have more than one category that only differ in their upper/lowercasing.

See the difference between these two:

- https://github.com/github/feedback/discussions/categories/discussions-feedback?discussions_q=category%3A%22Discussions+Feedback%22+downvote
- https://github.com/github/feedback/discussions/categories/discussions-feedback?discussions_q=category%3A%22Discussions+FEEDBACK%22+downvote

For comparison, the `label:` parameter is case-insensitive, as it should be.

### `category:`

![image](https://user-images.githubusercontent.com/6379424/120875479-3e126c00-c5d6-11eb-89af-bad6f249591e.png)

![image](https://user-images.githubusercontent.com/6379424/120875394-d6f4b780-c5d5-11eb-81b4-1ba8b3e83810.png)

### `label:`

![image](https://user-images.githubusercontent.com/6379424/120875353-a745af80-c5d5-11eb-81c9-d720cdbf4a65.png)

![image](https://user-images.githubusercontent.com/6379424/120875378-bdec0680-c5d5-11eb-8b7b-68cd0624abb5.png)

As with the `label:` parameter, the `category:` parameter should be case-insensitive.

This also affects the GraphQL API.

**Comment by laymonage** [comment]
<details>

<summary>

~~Related (fixed):~~
</summary>

When using the `category:` and `repo:` parameters, it seems that you need to use lowercase for the `repo:` parameter.

- correct case, with category (:x:  **NOT FOUND**):
  https://github.com/search?type=discussions&q=repo%3AAntoineSoetewey%2Fstatsandr+category%3AGeneral+in%3Atitle+blog%2Frunning-pace-calculator%2F
- lowercase, with category (:heavy_check_mark: **FOUND**):
  https://github.com/search?type=discussions&q=repo%3Aantoinesoetewey%2Fstatsandr+category%3AGeneral+in%3Atitle+blog%2Frunning-pace-calculator%2F
- correct case, without category (:heavy_check_mark: **FOUND**):
  https://github.com/search?type=discussions&q=repo%3AAntoineSoetewey%2Fstatsandr+in%3Atitle+blog%2Frunning-pace-calculator%2F
- lowercase, with category (:heavy_check_mark: **FOUND**):
  https://github.com/search?type=discussions&q=repo%3Aantoinesoetewey%2Fstatsandr+in%3Atitle+blog%2Frunning-pace-calculator%2F

As demonstrated, if both parameters are used, no results are returned when the `repo:` parameter is not in lowercase.
</details>

**Comment by Bestra** [comment]
Thanks for the great writeup! I've added this to our backlog.

**Comment by antn** [comment, marked as answer]
Thanks for reporting this! I've shipped a fix for this, so the category should be case insensitive now.
